packages/e2e/src/specs/contacts/new-contact.e2e.ts

```diff
@@ -3,123 +3,302 @@
  * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
  */
 
-import NavigationTabs from "../../page-objects/tabs.page"
-import ContactsPage from "../../page-objects/contacts.page"
-import ModalGeneralPage from "../../page-objects/modal-general.page"
-import ModalContactsPage from "../../page-objects/modal-contacts.page"
+import navigationTabs from "../../page-objects/tabs.page"
+import contactsPage from "../../page-objects/contacts.page"
+import modalGeneralPage from "../../page-objects/modal-general.page"
+import modalContactsPage from "../../page-objects/modal-contacts.page"
+
+import { contacts } from "../../test-data/contacts-only-test-data"
+import { waitForClickableAndClick } from "../../helpers/general.helper"
 
 describe("Add, edit, delete single contact scenarios", () => {
   before(async () => {
     // Waiting for device connected through USB
-    await browser.executeAsync((done) => {
-      setTimeout(done, 13000)
-    })
-    await ModalGeneralPage.clickUpdateAvailableModalCloseButton()
-    await NavigationTabs.clickContactsTab()
-    await browser.executeAsync((done) => {
-      setTimeout(done, 1000)
-    })
+    await browser.pause(10000)
+    await modalGeneralPage.clickUpdateAvailableModalCloseButton()
+    await navigationTabs.clickContactsTab()
   })
 
   it("Should press new contact button, add contact data and save it", async () => {
-    const newContactButton = await ContactsPage.newContactButton
-    await expect(newContactButton).toBeDisplayed()
-    await newContactButton.click()
+    const newContactButton = await contactsPage.newContactButton
+    await waitForClickableAndClick(newContactButton)
     await expect(newContactButton).toBeDisabled()
 
-    await ContactsPage.insertTextToFirstNameInput("Henryk")
-    await ContactsPage.insertTextToLastNameInput("Zaskroniec")
-    await ContactsPage.insertTextToPrimaryPhoneNumberInput("602900000")
-    await ContactsPage.insertTextToSecondaryPhoneNumberInput("510100100")
-    await ContactsPage.insertTextToAddressFirstLineInput(
-      "Al. Ujazdowskie 100/19"
+    await contactsPage.insertTextToFirstNameInput(contacts[1].firstName)
+    await contactsPage.insertTextToLastNameInput(contacts[1].lastName)
+    await contactsPage.insertTextToPrimaryPhoneNumberInput(
+      contacts[1].primaryNumber
```

Comment:
You can declare `const contactToAdd = contacts[1]` like you did in test below. A similar approach can be used for `storedContacts[0]`, `contacts[2]` etc.

Reply:
https://github.com/mudita/mudita-center/pull/1201/commits/08252851ec674568c6c4262bcebf0327ead77165